base_layer/wallet/src/output_manager_service/storage/sqlite_db/output_sql.rs

```diff
@@ -371,106 +372,110 @@ impl OutputSql {
+                            OutputStatus::try_from(x.status).unwrap(),
+                            EncumberedToBeReceived | ShortTermEncumberedToBeReceived | UnspentMinedUnconfirmed
+                        ) && x.source != OutputSource::Coinbase as i32
+                    })
+                    .map(|x| MicroTari(x.amount as u64))
+                    .sum(),
+                pending_outgoing_balance: amounts
+                    .iter()
+                    .filter(|x| {
+                        matches!(
+                            OutputStatus::try_from(x.status).unwrap(),
+                            EncumberedToBeSpent | ShortTermEncumberedToBeSpent | SpentMinedUnconfirmed
+                        )
+                    })
+                    .map(|x| MicroTari(x.amount as u64))
+                    .sum(),
+            },
+            // with height
+            Some(height) => {
+                let height = height as i64;
+                Balance {
+                    available_balance: amounts
+                        .iter()
+                        .filter(|x| {
+                            matches!(OutputStatus::try_from(x.status).unwrap(), Unspent) &&
+                                x.script_lock_height <= height &&
+                                x.maturity <= height
+                        })
+                        .map(|x| MicroTari(x.amount as u64))
+                        .sum(),
+                    time_locked_balance: amounts
+                        .iter()
+                        .filter(|x| {
+                            matches!(OutputStatus::try_from(x.status).unwrap(), Unspent) &&
+                                x.script_lock_height > height &&
+                                x.maturity > height
+                        })
+                        .map(|x| MicroTari(x.amount as u64))
+                        .sum1(),
+                    pending_incoming_balance: amounts
+                        .iter()
+                        .filter(|x| {
+                            matches!(
+                                OutputStatus::try_from(x.status).unwrap(),
+                                EncumberedToBeReceived | ShortTermEncumberedToBeReceived | UnspentMinedUnconfirmed
+                            ) && x.source != OutputSource::Coinbase as i32
+                        })
+                        .map(|x| MicroTari(x.amount as u64))
+                        .sum(),
+                    pending_outgoing_balance: amounts
+                        .iter()
+                        .filter(|x| {
+                            matches!(
+                                OutputStatus::try_from(x.status).unwrap(),
+                                EncumberedToBeSpent | ShortTermEncumberedToBeSpent | SpentMinedUnconfirmed
+                            )
+                        })
+                        .map(|x| MicroTari(x.amount as u64))
+                        .sum(),
+                }
+            },
```

Comment:
This logic here, needs to happen in SQL. 
This is a regression to how it was before we fixed the performance issues here. 
If you have many outputs, this will be slow. 
If you compare speed, SQL >> RUST. 
There should be no filtering done in RUST here.